### ISSUE TITLE
Compiler warns for a field which initialized within constructor inside try block

### ISSUE THREAD

**Issue by buyaa-n** [dotnet/roslyn#42026]
Compiler warns for a field which initialized within constructor inside try block
**Steps to Reproduce**:
```csharp
class Program
{
    private string _value;
    public Program() { // warns here: Non nullable field '_value' uninitialized
        try
        {
            _value = string.Empty;
        }
        catch (Exception){ }
    }
}
```
**Expected Behavior**:
Shouldn't warn

**Comment by RikkiGibson** [comment]
I think this is by-design because it is possible for the constructor to return normally without the _value being initialized (due to the catch block).

Compare to this similar program that has no warning:

```csharp
class Program
{
    private string _value;
    public Program() { // warns here: Non nullable field '_value' uninitialized
        try
        {
            _value = string.Empty;
        }
        finally { }
    }
}
```

**Comment by buyaa-n** [comment]
You are right, its not warning if it throws on catch, thanks! Closing
